docs/advanced-features/instrumentation.md

```diff
@@ -0,0 +1,36 @@
+---
+description: Learn how to instrument your Next.js app.
+---
+
+> **Note**: This feature is experimental. To use it, you must explicitly opt in by defining `experimental.instrumentationHook = true;` in your `next.config.js`.
+
+# `instrumentation.ts`
+
+If you export a function named `register` from this file, we will call that function whenever a new server instance is bootstrapped.
+When your `register` function is deployed, it will be called continuously on each cold boot (but exactly once in each environment).
+
+Sometimes, it may be useful to import a file in your code because of the side effects it will cause. For example, you might import a file that defines a set of global variables, but never explicitly use the imported file in your code. You would still have access to the global variables the package has declared.
+
+You can import files with side effects in `instrumentation.ts`, which you might want to use in your `register` function as demonstrated in the following example:
```

Comment:
add a mention that this file will get compiled for both node+ the edge, hence the usage of process.env for code pslitting